### DISCUSSION TITLE
collect() with chunk()

### DISCUSSION THREAD

**Discussion by rus-ik** [laravel/framework#38255] [Q&A] [answered]
- Laravel Version: 8.52.0
- PHP Version: 7.4.15
- Database Driver & Version:

### Description:
Shouldn't the collection of the 1st chunk (in the output below) return something like
...
all: [
       1 => 1,
       2 => 2,
     ],

?

>>> collect([1,2,3,4,5,6])->chunk(2)
=> Illuminate\Support\Collection {#5086
     all: [
       Illuminate\Support\Collection {#5103
         all: [
           1,
           2,
         ],
       },
       Illuminate\Support\Collection {#5096
         all: [
           2 => 3,
           3 => 4,
         ],
       },
       Illuminate\Support\Collection {#5098
         all: [
           4 => 5,
           5 => 6,
         ],
       },
     ],
   }

### Steps To Reproduce:
echo 'collect([1,2,3,4,5,6])->chunk(2)' | php artisan tinker

**Comment by thoasty-dev** [comment, marked as answer]
Array indexes in php are zero-based, which means the first element has an index of 0.

So the expected output is

```[0 => 1, 1 => 2]```

which is equivalent to your output

```[1, 2]```
